### PR TITLE
Je 56113

### PR DESCRIPTION
**Pull request by lazarenkoalexey**

### CODE REVIEW

File: manifest.jps [hunk 1]
```diff
@@ -199,10 +200,12 @@ actions:
       fallbackToX1: ${globals.fallbackToX1}
       test: ${globals.test}
       clientVersion: ${globals.version_acme-sh:}
+      envAppid: ${globals.envAppid}
   
   callScript:
     script: |
-      var j = jelastic, resp = j.dev.scripting.Eval(appid, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}', {action:'${this.action}'});
+      var appidValue = api.dev.apps.CreatePersistence ? "${globals.envAppid}" : appid;
+      var j = jelastic, resp = j.dev.scripting.Eval(appidValue, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}', {action:'${this.action}'});
```
Comment (SlavaKatiukha) [inline manifest.jps:208]:
```suggestion
      var resp = api.dev.scripting.Eval(appidValue, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}', {action:'${this.action}'});
```

File: manifest.jps [hunk 2]
```diff
@@ -255,7 +258,8 @@ actions:
   
   delete:
     script: |
-      jelastic.dev.scripting.DeleteScript('${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}');
+      var appidValue = api.dev.apps.CreatePersistence ? "${globals.envAppid}" : appid;
```
Comment (SlavaKatiukha) [inline manifest.jps:261]:
```suggestion
      var targetAppid = api.dev.apps.CreatePersistence ? "${globals.envAppid}" : appid;
```

File: manifest.jps [hunk 3]
```diff
@@ -255,7 +258,8 @@ actions:
...
+      jelastic.dev.scripting.DeleteScript(appidValue, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}');
```
Comment (SlavaKatiukha) [inline manifest.jps:262]:
```suggestion
      api.dev.scripting.DeleteScript(targetAppid, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}');
```

File: manifest.jps [hunk 4]
```diff
@@ -275,6 +279,7 @@ actions:
       
   createScript:
     script: |
-      return jelastic.dev.scripting.CreateScript("${this.scriptName}", "js", "${this.code:}")
+      var appidValue = api.dev.apps.CreatePersistence ? "${globals.envAppid}" : appid;
```
Comment (SlavaKatiukha) [inline manifest.jps:282]:
```suggestion
      var targetAppid = api.dev.apps.CreatePersistence ? "${globals.envAppid}" : appid;
```

File: manifest.jps [hunk 5]
```diff
@@ -275,6 +279,7 @@ actions:
...
+      return jelastic.dev.scripting.CreateScript(appidValue, session, "${this.scriptName}", "js", "${this.code:}")
```
Comment (SlavaKatiukha) [inline manifest.jps:283]:
```suggestion
      return api.dev.scripting.CreateScript(targetAppid, session, "${this.scriptName}", "js", "${this.code:}")
```

File: scripts/ssl-manager.js [hunk 6]
```diff
@@ -58,6 +58,7 @@ function SSLManager(config) {
         CONFIGURE = "configure",
         Random = com.hivext.api.utils.Random,
         isAddedEnvDomain = false,
+        scriptsAppid = "",
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:61]:
```suggestion
        targetAppid,
```

File: scripts/ssl-manager.js [hunk 7]
```diff
@@ -75,6 +76,8 @@ function SSLManager(config) {
     config = config || {};
     session = config.session || "";
 
+    scriptsAppid = api.dev.apps.CreatePersistence ? config.envAppid : appid;
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:79]:
```suggestion
    targetAppid = api.dev.apps.CreatePersistence ? config.envAppid : appid;
```

File: scripts/ssl-manager.js [hunk 8]
```diff
@@ -522,7 +525,7 @@ function SSLManager(config) {
             sCode = nodeManager.getCSScriptCode();
 
         me.logAction("AutoPatchLEScriptRestore");
-        return jelastic.dev.scripting.CreateScript(config.scriptName, "js", sCode);
+        return api.dev.scripting.CreateScript(scriptsAppid, session, config.scriptName, "js", sCode);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:528]:
```suggestion
        return api.dev.scripting.CreateScript(targetAppid, session, config.scriptName, "js", sCode);
```

File: scripts/ssl-manager.js [hunk 9]
```diff
@@ -965,11 +968,10 @@ function SSLManager(config) {
             resp = getScript(scriptingScriptName);
             if (resp.result == Response.OK) {
                 //delete the script if it already exists
-                api.dev.scripting.DeleteScript(appid, session, scriptingScriptName);
+                api.dev.scripting.DeleteScript(scriptsAppid, session, scriptingScriptName);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:971]:
```suggestion
                api.dev.scripting.DeleteScript(targetAppid, session, scriptingScriptName);
```

File: scripts/ssl-manager.js (hunk 8 above)
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:528]:
```suggestion
        return api.dev.scripting.CreateScript(targetAppid, session, config.scriptName, "js", sCode);
```

File: scripts/ssl-manager.js (hunk 9 above)
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:971]:
```suggestion
                api.dev.scripting.DeleteScript(targetAppid, session, scriptingScriptName);
```

File: scripts/ssl-manager.js [hunk 10]
```diff
@@ -965,11 +968,10 @@ function SSLManager(config) {
...
             }
             //create a new script
-            resp = api.dev.scripting.CreateScript(appid, session, scriptingScriptName, "js", scriptBody);
-
+            resp = api.dev.scripting.CreateScript(scriptsAppid, session, scriptingScriptName, "js", scriptBody);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:974]:
```suggestion
            resp = api.dev.scripting.CreateScript(targetAppid, session, scriptingScriptName, "js", scriptBody);
```

File: scripts/ssl-manager.js [hunk 11]
```diff
@@ -1021,7 +1023,7 @@ function SSLManager(config) {
         if (action) params.action = action;
         params.fallbackToX1 = config.fallbackToX1;
 
-        var resp = jelastic.dev.scripting.Eval(config.scriptName, params);
+        var resp = jelastic.dev.scripting.Eval(scriptsAppid, session, config.scriptName, params);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:1026]:
```suggestion
        var resp = api.dev.scripting.Eval(targetAppid, session, config.scriptName, params);
```

File: scripts/ssl-manager.js [hunk 12]
```diff
@@ -2102,7 +2104,7 @@ function SSLManager(config) {
     }
 
     function getScript(name) {
-        return api.dev.scripting.GetScript(appid, session, name);
+        return api.dev.scripting.GetScript(scriptsAppid, session, name);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:2107]:
```suggestion
        return api.dev.scripting.GetScript(targetAppid, session, name);
```

File: scripts/ssl-manager.js [hunk 13]
```diff
@@ -2118,7 +2120,7 @@ function SSLManager(config) {
 
     function log(message) {
         if (jelastic.marketplace && jelastic.marketplace.console && message) {
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:2122]:
```suggestion
        if (api.marketplace && api.marketplace.console && message) {
```

File: scripts/ssl-manager.js [hunk 14]
```diff
@@ -2118,7 +2120,7 @@ function SSLManager(config) {
...
-            return jelastic.marketplace.console.WriteLog(appid, session, message);
+            return jelastic.marketplace.console.WriteLog(message);
```
Comment (SlavaKatiukha) [inline scripts/ssl-manager.js:2123]:
```suggestion
            return api.marketplace.console.WriteLog(message);
```

**Review by SlavaKatiukha — CHANGES_REQUESTED** [review]

File: manifest.jps [hunk 15]
```diff
@@ -50,7 +50,7 @@ targetNodes:
     - jenkins2
 
 homepage: https://github.com/jelastic-jps/lets-encrypt
-baseUrl: https://raw.githubusercontent.com/jelastic-jps/lets-encrypt/master
+baseUrl: https://raw.githubusercontent.com/jelastic-jps/lets-encrypt/JE-56113
```
Comment (SlavaKatiukha) [inline manifest.jps:53]: don't forget to change after testing

File: manifest.jps [hunk 16]
```diff
@@ -199,10 +200,12 @@ actions:
       fallbackToX1: ${globals.fallbackToX1}
       test: ${globals.test}
       clientVersion: ${globals.version_acme-sh:}
+      envAppid: ${globals.envAppid}
   
   callScript:
     script: |
-      var j = jelastic, resp = j.dev.scripting.Eval(appid, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}', {action:'${this.action}'});
+      var targetAppid = api.dev.apps.CreatePersistence ? "${globals.envAppid}" : appid;
+      var resp = api.dev.scripting.Eval(appidValue, session, '${env.envName}-${targetNodes.nodeGroup}-${globals.scriptSufix}', {action:'${this.action}'});
```
Comment (SlavaKatiukha) [inline manifest.jps:208]: appidValue?

**Review by SlavaKatiukha — CHANGES_REQUESTED** [review]

File: manifest.jps (hunk 16 above)
Comment (lazarenkoalexey) [reply manifest.jps]: fixed

**Review by SlavaKatiukha — APPROVED** [review]

**Review by SlavaKatiukha — APPROVED** [review]

**Pull request merged**